lib/layer.js

```diff
@@ -107,24 +118,33 @@ Layer.prototype.handle_request = function handle(req, res, next) {
 
 Layer.prototype.match = function match(path) {
   var match
+  var checkPath
 
   if (path != null) {
     // fast path non-ending match for / (any path matches)
-    if (this.regexp.fast_slash) {
+    if (this.fastSlash) {
       this.params = {}
       this.path = ''
+      this.matchedPath = this.paths[0]
       return true
     }
 
     // fast path for * (everything matched in a param)
-    if (this.regexp.fast_star) {
+    if (this.fastStar) {
       this.params = {'0': decode_param(path)}
       this.path = path
+      this.matchedPath = this.paths[0]
       return true
     }
 
     // match the path
-    match = this.regexp.exec(path)
+    for (var i = 0; i < this.paths.length; i++) {
+      checkPath = this.paths[i]
+      if (match = checkPath.regexp.exec(path)) {
+        this.matchedPath = checkPath
+        break
+      }
+    }
   }
 
   if (!match) {
```

Comment:
I think `this.matchedPath` should be reset inside this conditional as well, otherwise it will look like there is a matched path when none of the paths matched, but they did in a previous attempt.

Reply:
Updated. Is there a way to test this?

Reply:
There should be by creating a specific order of routes and running though them. I can help make one if you need, perhaps this weekend